base/cvd/cuttlefish/host/libs/web/cas/cas_downloader.cpp

```diff
@@ -0,0 +1,296 @@
+//
+// Copyright (C) 2024 The Android Open Source Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "host/libs/web/cas/cas_downloader.h"
+
+#include <android-base/strings.h>
+
+#include <cstddef>
+#include <cstdio>
+#include <map>
+#include <memory>
+#include <optional>
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "common/libs/utils/files.h"
+#include "common/libs/utils/result.h"
+#include "common/libs/utils/json.h"
+#include "common/libs/utils/subprocess.h"
+#include "android-base/logging.h"
+#include "json/value.h"
+
+namespace cuttlefish {
+
+namespace {
+
+std::set<std::string> GetSupportedFlags(const std::string& go_binary_path) {
+  std::set<std::string> flags;
+
+  Command command(go_binary_path);
+  command.AddParameter("-help");
+  std::string out;
+  std::string err;
+  RunWithManagedStdio(std::move(command), nullptr, &out, &err);
+  // "casdownloader -help" outputs to stderr.
+  std::vector<std::string> lines = android::base::Tokenize(err, "\n");
+  for (std::string& line : lines) {
+    // Sample help output line: "  -version=false: Print version information"
+    if (line.find("  -") != std::string::npos) {
+      size_t start = line.find("  -") + 3;
+      size_t end = line.find('=', start);
```

Comment:
1. Added `if (end > start)` test to avoid adding an empty flag (not sensible).
2. Due to the preceding if-test, start should always have a valid value that's not `std::string::npos`. If line is "  -", than start will be 3 and the find() should work. Did I get this wrong?